### PR TITLE
Add new function `-shuffle`

### PR DESCRIPTION
**Pull request by cireu**
See https://github.com/magnars/dash.el/issues/283

### CODE REVIEW

File: dash.el [hunk 1]
```diff
@@ -2404,6 +2404,22 @@ if the first element should sort before the second."
...
+Shuffling the list by using Fisher-Yates shuffle algorithm.
+See https://en.wikipedia.org/wiki/Fisher-Yates_shuffle for more details."
+  (let ((source (copy-sequence list))
+        (random-nums (->> (length list)
+                          (number-sequence 1)
+                          (-map #'random)
+                          nreverse))
+        result)
+    (--each random-nums
+      (-let [(front (target . rest)) (-split-at it source)]
```
Comment (Fuco1) [inline dash.el]: Nice, even better than what I had :)

**Comment by cireu** [comment]
Why CI test failed? Flycheck also complains that `dash--match`'s definition is void. But `dash--match` was defined at line 1849 @Fuco1

**Comment by Fuco1** [comment]
You can execute `(random "dash")` to set the seed for the random generator, then it should always return the same sequence

```elisp
(progn
  (random "dash")
  (message "first %s" (-map (lambda (x) (mod x 100)) (-iterate #'random 10 10)))
  (random "dash") ;; comment out to see a different sequence
  (message "second %s" (-map (lambda (x) (mod x 100)) (-iterate #'random 10 10))))
```

**Comment by cireu** [comment]
I  know, but it seems that `dash--match` is at line 1849, `-let` is at line 1916. And `-shuffle` is at line 2407

**Comment by Fuco1** [comment]
@cireu I'm not sure then.  Probably you will have to expand the code or rewrite it without the `-let` macro :(

I've run into this problem before and I couldn't solve it but I should give it more time because it's quite annoying.

**Pull request closed without merging**